File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        
        HashMap<String, Integer> type = new HashMap<>();
        
        for(int i=0; i<clothes.length; i++){
            type.put(clothes[i][1], type.getOrDefault(clothes[i][1], 0) + 1);
        }
        
        int answer = 1;
        
        for(int value : type.values()){
            answer *= (value+1);
        }
   
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    
    var type = new Map();
    
    for(let i=0; i<clothes.length; i++){
        if(type.has(clothes[i][1])){
            type.set(clothes[i][1], type.get(clothes[i][1])+1);
        }else{
            type.set(clothes[i][1], 1);
        }
    }
    
    var answer = 1;
    
    type.forEach((value, key) => {
        answer *= (value+1);
    })
    
    return answer-1;
}